### PR TITLE
LoadBalancer in BackendTrafficPolicy

### PR DESCRIPTION
**Pull request by arkodg**
Fixes: https://github.com/envoyproxy/gateway/issues/1105

### CODE REVIEW

File: api/v1alpha1/loadbalancer_types.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// "Random",
+	// "RoundRobin",
+	//
+	// +unionDiscriminator
+	Type LoadBalancerType `json:"type"`
+	// ConsistentHash defines the configuration when the load balancer type is
+	// set to ConsistentHash
+	//
+	// +optional
+	ConsistentHash *ConsistentHash `json:"consistentHash,omitempty"`
```
Comment (zirain) [inline api/v1alpha1/loadbalancer_types.go:24]: haha, this's a good place to try XValidation

Comment (arkodg) [reply api/v1alpha1/loadbalancer_types.go:24]: Yes

Comment (arkodg) [reply api/v1alpha1/loadbalancer_types.go:24]: Can we do it as a follow up ?

Comment (zirain) [reply api/v1alpha1/loadbalancer_types.go:24]: sure, it's not a block

Comment (zhaohuabing) [inline api/v1alpha1/loadbalancer_types.go:24]: What happens if the type is `ConsistentHash` and the `ConsistentHash` is nil?

Comment (arkodg) [reply api/v1alpha1/loadbalancer_types.go:24]: this is what @zirain is referring to in https://github.com/envoyproxy/gateway/pull/2063#discussion_r1372418714
we can add CEL Validation to proactively check this

**Review by zirain — APPROVED** [review]

File: api/v1alpha1/loadbalancer_types.go (hunk 1 above)
Comment (zirain) [reply api/v1alpha1/loadbalancer_types.go:24]: can you open a ticket for this?

Comment (arkodg) [reply api/v1alpha1/loadbalancer_types.go:24]: raised https://github.com/envoyproxy/gateway/issues/2066

**Pull request merged**